src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null");
+        }
+        if (user.getPassword().length() < MINIMUM_LENGTH) {
+            throw new RegistrationException("Not valid password " + user.getPassword()
+                    + ".Password cannot be less than " + MINIMUM_LENGTH + " characters");
+        }
+        if (user.getLogin().length() < MINIMUM_LENGTH) {
+            throw new RegistrationException("Not valid login " + user.getLogin()
+                    + ".Login cannot be less than " + MINIMUM_LENGTH + " characters");
```

Comment:
The validation for the login's minimum length seems to be an additional requirement that is not described in the task. Please ensure that this check is necessary according to the task requirements. If it is not, this check should be removed.